README.md

```diff
@@ -1,7 +1,111 @@
-### ReactiveFeedback
+# ReactiveFeedback
 
-Unideraction reactive architecture. This is [ReactiveSwift](https://github.com/ReactiveCocoa/ReactiveSwift) implemetation of the [RxFeedback](https://github.com/kzaher/RxFeedback)
+Unidirectional reactive architecture. This is a [ReactiveSwift](https://github.com/ReactiveCocoa/ReactiveSwift) implemetation of the [RxFeedback](https://github.com/kzaher/RxFeedback)
 
-### Documentation
+## Documentation
 
-* TBD
+![](diagrams/ReactiveFeedback.jpg)
+
+### Motivation
+
+Requirements for iOS apps have become huge. Our code has to manage a lot of state e.g. server responses, cached data, UI state, routing etc. Some may say that Reactive Programming can help us a lot, but in the wrong hands, it can make even more harm to your code base.
+
+The goal of this library is to provide a simple and intuitive approach to designing reactive state machines.
+
+### Core Concepts
+
+##### State 
+
+`State` the single source of truth. It represents a state of your system. Usually a plain swift type (does not contain any ReactiveSwift primitives) to the point that it can be saved on disk. The only way to change a `State` is to emit an `Event`
+
+```swift
+enum State {
+    case loading
+    case loaded([Item])
+}
+```
+
+##### Event
+
+Represent all possible events that can happen in your system which can transition to a new value of the `State`
+
+```swift
+enum Event {
+    case startLoadingNextPage
+    case loaded([Movie])
+    case failed(Error)
+    case retry
+}
+```
+
+##### Reducer 
+
+Is a pure function of `(State, Event) -> State`. This is the only place where the `State` can be changed.
+
+```swift
+func reduce(state: State, event: Event) -> State {
+    switch event {
+    case .startLoadingNextPage:
+        return .paging(context: state.context)
+    case .response(let batch):
+        var copy = state.context
+        copy.batch = batch
+        copy.movies += batch.results
+        return .loadedPage(context: copy)
+    case .failed(let error):
+        return .error(error: error, context: state.context)
+    case .retry:
+        return .retry(context: state.context)
+    }
+}
+```
+
+##### Feedback
+
+Represents effects that may happen in your system that somehow can mutate the `State`, e.g (UI events, Networking, DB fetches, timers, Bluetooth ...). Essentially it's a pure function of `Signal<State, NoError> -> Signal<Event, NoError>`. Feebacks don't directly mutate the state, they only emit events, which cause the state to change in the reducer.
```

Comment:
"While `State` represents where the system *is* at a given time, `Event` *represents* a state change, and `Reducer` is the pure function that *enacts* the event causing the state to change, there is not as of yet any type to *decide* which event should take place given a particular current state. That's the job of the `Feedback`. It's essentially a "processing engine", listening to changes in the current state and emitting the corresponding next event to take place. It's represented by a pure function of `Signal<State, NoError>` to `Signal<Event, NoError>`. Feebacks don't directly mutate states. Instead, they only emit events which then cause states to change in reducers."